packages/components/src/navigation/item/index.js

```diff
@@ -63,14 +63,15 @@ export default function NavigationItem( props ) {
 		onClick( event );
 	};
 	const icon = isRTL() ? chevronLeft : chevronRight;
-	const baseProps = isText
+	const baseProps = children ? props : { ...props, onClick: undefined };
```

Comment:
I might be missing something, but if `isText` is `false` and `children` is `true`, couldn't we still have 2 `onClick` handlers registered?

Reply:
I don't think so 🤔 We don't apply the `props` on the `children` automatically. So in this case `NavigationItemBase` will get an `onClick`˛handler but nothing else. You can still put an `onClick` handler on your `children` in the userland code. But that would mean you are using the component wrong.

Correct way is to add your `onClick` to the `NavigationItem`:
```js
<NavigationItem onClick={() => alert('Hello')}>
<div>There is only one event click handler</div>
</NavigationItem>
```

Wrong way

```js
<NavigationItem onClick={() => alert('Hello')}>
<div onClick={() => alert('Hello children!')}>There is only one event click handler</div>
</NavigationItem>
```

Note, when we have `children` then the `itemProps` is not used at all.

Reply:
> Note, when we have children then the itemProps is not used at all

Ahhh I think this is what's confusing me. Why use `isText` at all if we can infer the same state from the presence of `children`?

Reply:
We need to make sure `ItemUI` is acting as a text in case of `isText`, otherwise we turn it into a `button`.

`isText = true` = We pass rest props, so it's up to the user to add extra props
`isText = false` = We default to a clickable navigation button

Reply:
Oh I see it now 🤦 . I'm tracking. Thanks for the explanation @david-szabo97!